src/main/kotlin/pl/newbies/event/domain/service/EventService.kt

```diff
@@ -127,4 +148,22 @@ class EventService {
             nameWithExtension = nameWithExtension,
         )
     }
+
+    private fun getVisibilityQuery(visibilityIn: List<Event.Visibility>, requesterId: String?): Op<Boolean> {
+        var query: Op<Boolean> = Op.FALSE
+
+        if (visibilityIn.contains(Event.Visibility.PUBLIC)) {
+            query = query or (Events.visibility eq Event.Visibility.PUBLIC)
+        }
+        if (requesterId != null) {
+            if (visibilityIn.contains(Event.Visibility.INVISIBLE)) {
+                query = query or ((Events.visibility eq Event.Visibility.INVISIBLE) and (Events.author eq requesterId))
+            }
+            if (visibilityIn.contains(Event.Visibility.PRIVATE)) {
+                query = query or ((Events.visibility eq Event.Visibility.PRIVATE) and (Events.author eq requesterId))
+            }
```

Comment:
yo, this is for list or single event

Reply:
I've changed function name, that should clarify this question (It's meant to be used in list)